### PR TITLE
mem-hotplug: fix target pod requests calculation with hugepages

### PR DESCRIPTION
**Pull request by acardace**
This fixes a bug where you cannot hotplug successfully when using hugepages due to virt-launcher being tricked into thinking that the target pod does not have enough memory to accomodate the new VMI with additional hotplugged memory.

### What this PR does
Before this PR:
Doing memory hotplug with hugepages gets the hotplug migration stuck indefinitely. 

After this PR:
Doing memory hotplug with hugepages works as expected.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-controller/watch/migration_test.go [hunk 1]
```diff
@@ -578,6 +580,13 @@ var _ = Describe("Migration watcher", func() {
 					Name: "compute", State: k8sv1.ContainerState{Running: &k8sv1.ContainerStateRunning{}},
 				}}
 
+				expectedRequestsLabel := targetPod.Spec.Containers[0].Resources.Requests.Memory().DeepCopy()
+				if hugepages != nil {
+					resourceName := k8sv1.ResourceName(k8sv1.ResourceHugePagesPrefix + hugepages.PageSize)
+					targetPod.Spec.Containers[0].Resources.Requests[resourceName] = guestMemory
+					expectedRequestsLabel.Add(guestMemory)
+				}
```
Comment (xpivarc) [inline pkg/virt-controller/watch/migration_test.go:588]: Nit: Please move this to the argument of the `func`

**Review by xpivarc — COMMENTED** [review]
Can we have or extend existing test?

**Comment by acardace** [comment]
@xpivarc added functional test

File: pkg/virt-controller/watch/migration_test.go (hunk 1 above)
Comment (acardace) [reply pkg/virt-controller/watch/migration_test.go]: done.

**Review by fossedihelm — APPROVED** [review]
Looks good to me.

**Comment by xpivarc** [comment]
@acardace Flake detected

File: tests/hotplug/memory.go [hunk 2]
```diff
@@ -158,7 +159,10 @@ var _ = Describe("[sig-compute][Serial]Memory Hotplug", decorators.SigCompute, d
 			Expect(compute).NotTo(BeNil(), "failed to find compute container")
 			reqMemory = compute.Resources.Requests.Memory().Value()
 			Expect(reqMemory).To(BeNumerically(">=", newGuestMemory.Value()))
-		})
+		},
+			Entry("with a common VM"),
+			Entry("with 2Mi pagesize hugepages VM", decorators.RequiresHugepages2Mi, libvmi.WithHugepages("2Mi")),
```
Comment (xpivarc) [inline tests/hotplug/memory.go:164]: This option is passed but there is no argument in the table's function

Comment (acardace) [reply tests/hotplug/memory.go:164]: darn, I'll add it.

Comment (acardace) [reply tests/hotplug/memory.go:164]: Fixed it.

Comment (fossedihelm) [reply tests/hotplug/memory.go:164]: :man_facepalming: <- on me

**Review by xpivarc — COMMENTED** [review]
/approve

**Review by fossedihelm — APPROVED** [review]

**Comment by enp0s3** [comment]
@acardace @xpivarc Hi. When hupe-pages are used as a backed for guest virtual memory, and we do mem. hotplug, shouldn't we care only about enough HP resources on the target pod?

Why do we add the target pod memory request to the HP req?

**Pull request merged**